p2p/libp2p/factory/peerDiscovererFactory.go

```diff
@@ -0,0 +1,58 @@
+package factory
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/p2p/libp2p/discovery"
+)
+
+type peerDiscovererFactory struct {
+	p2pConfig config.P2PConfig
+}
+
+// NewPeerDiscovererFactory creates a new instance of peer discovery factory
+func NewPeerDiscovererFactory(pConfig config.P2PConfig) *peerDiscovererFactory {
+	return &peerDiscovererFactory{
```

Comment:
maybe check the config for nil and also return an error ?

Reply:
config could be verified at NewPeerDiscovererFactory time.

Reply:
pConfig can not be nil here as it has been pass as value not pointer

Reply:
> config could be verified at NewPeerDiscovererFactory time.

The checking logic will be duplicated in each constructor. I would avoid duplicate code in this case